plugins/gopacket/run.go

```diff
@@ -0,0 +1,247 @@
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+
+	s := <-sig
+	log.Warningf("Received a termination signal: %v", s)
+	cancel()
+}
+
+func (a *Agent) startReadOutputItems(ctx context.Context, outputItems chan *api.OutputChannelItem) {
+	log.Info("Starting to read output items")
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case item := <-outputItems:
+			go func() {
+				log.Infof("Output item received: connection info: %+v", item.ConnectionInfo)
+				if a.shouldFilterTelemetry(item) {
+					return
+				}
+				telemetry, err := a.createTelemetry(item)
+				if err != nil {
+					log.Errorf("Failed to create telemetry: %v", err)
+					return
+				}
+
+				params := operations.NewPostTelemetryParams().WithBody(telemetry)
+
+				_, err = a.apiClient.Operations.PostTelemetry(params)
+				if err != nil {
+					log.Errorf("Failed to post telemetry: %v", err)
+					return
+				}
+				log.Info("Telemetry has been sent")
+			}()
+		}
+	}
+}
+
+// for each connection we will get the pod to pod communication (maybe twice if they are on different nodes) and the pod to service communication.
+// the filtering logic is that we only send to API Clarity telemetries where the client is in a monitored namespace
+// and the destination is not a pod IP (service IP or external IP). Same behaviour as the wasm filter.
+func (a *Agent) shouldFilterTelemetry(item *api.OutputChannelItem) bool {
+	clientNamespace := a.podMonitor.GetPodNamespaceByIP(item.ConnectionInfo.ClientIP)
+	if !a.podMonitor.IsMonitoredNamespace(clientNamespace) {
+		// client pod is not on monitored namespace, ignore
+		return true
+	}
+	serverNamespace := a.podMonitor.GetPodNamespaceByIP(item.ConnectionInfo.ServerIP)
+
+	return serverNamespace != ""
+}
+
+func (a *Agent) createTelemetry(item *api.OutputChannelItem) (*models.Telemetry, error) {
+	requestHTTPPayload, ok := item.Pair.Request.Payload.(api.HTTPPayload)
+	if !ok {
+		return nil, fmt.Errorf("failed to convert request Payload into HTTPPayload")
+	}
+
+	request, ok := requestHTTPPayload.Data.(*http.Request)
+	if !ok {
+		return nil, fmt.Errorf("failed in type assertion to http.Request")
+	}
+
+	responseHTTPPayload, ok := item.Pair.Response.Payload.(api.HTTPPayload)
+	if !ok {
+		return nil, fmt.Errorf("failed to convert response Payload into HTTPPayload")
+	}
+
+	response, ok := responseHTTPPayload.Data.(*http.Response)
+	if !ok {
+		return nil, fmt.Errorf("failed in type assertion to http.Response")
+	}
+
+	reqBody, truncatedBodyReq, err := plugins.ReadBody(request.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read request body: %v", err)
+	}
+	resBody, truncatedBodyRes, err := plugins.ReadBody(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %v", err)
+	}
+
+	pathAndQuery := plugins.GetPathWithQuery(request.URL)
+
+	destinationNamespace := a.podMonitor.GetPodNamespaceByIP(item.ConnectionInfo.ServerIP)
+
```

Comment:
`a.podMonitor.GetPodNamespaceByIP(item.ConnectionInfo.ServerIP)` called both in `createTelemetry` and `shouldFilterTelemetry` it will be better to call it once and pass as a param into the functions